### ISSUE TITLE
RPGLE Linter - Error linting "RPG-source": Cannot read properties of undefined (reading 'type')

### ISSUE THREAD

**Issue by dennisnel** [halcyon-tech/vscode-rpgle#125]
Hi,

I'm working on source in a library on the IBM i.

I have the linting enabled and configured (MYSOURCELIB/VSCODE(RPGLINT.JSON)

As soon as I enable/configure  "NoUnreferenced" I get the error

"Error linting /MIDAS5149/QRPGSRC/RPGFREE.RPGLE: Cannot read properties of undefined (reading 'type')

I'm on v1.5.11 of Code for IBM i and v0.10.3 of RPGLE language tools

When I remove the "NoUnreferenced" line and restart the linter works.

![image](https://user-images.githubusercontent.com/69187463/189104540-abc39a04-bad8-4092-85d3-bb5c39bd6c47.png)

Here is a copy of my RPGLE.JSON
![image](https://user-images.githubusercontent.com/69187463/189104635-65eed87c-6a42-4143-afb6-fb9bd05758df.png)

![image](https://user-images.githubusercontent.com/69187463/189104662-02f9f83a-4f6e-4b2f-8bc0-ddb1318d6111.png)
![image](https://user-images.githubusercontent.com/69187463/189104692-9ddc5ac5-2768-4e03-a337-821931cf46c5.png)

**Comment by dennisnel** [comment]
I could also have missed something, but as soon as I make changes to the RPGLE.JSON file I need to open/close vscode for the linter to recognize the "new" config.

Is this "normal" or is there a way to refresh the new config without a open/close?

Thx

**Comment by worksofliam** [comment]
@dennisnel are you able to share the code for where the error is happening?

**Comment by worksofliam** [comment]
@dennisnel I will still need an example to understand what might be causing this issue :)

**Comment by dennisnel** [comment]
This was non-FREE RPGLE that I converted and then wanted to open it and get the linter errors.

[RPGFREE.rpgle.txt](https://github.com/halcyon-tech/vscode-rpgle/files/9527230/RPGFREE.rpgle.txt)

![image](https://user-images.githubusercontent.com/69187463/189136186-0f059610-ba24-4750-b784-22c558b3edef.png)

**Comment by worksofliam** [comment]
@dennisnel Ok, this is going to be very tricky.

You need to open each of these members individually and share with me the one where the outline view is not renderer (or causes the error).

I am able to get the outline view to show without the copybooks.

```
/INCLUDE QCPYSRC,CUSTMSTPF

//  -------------------------------------------------------------------
//  Procedure Prototypes
//  -------------------------------------------------------------------
/Include QCPYSRC,ERR03
/Include QCPYSRC,ERR11
/Include QCPYSRC,SPMIDASUTL

//  -------------------------------------------------------------------
//  Data Structures
//  -------------------------------------------------------------------
/INCLUDE QCPYSRC,PSDS
/INCLUDE QCPYSRC,RPGLECONST
```

**Comment by worksofliam** [comment]
@dennisnel We now provide a new way to provide debug information, which should help me track down your issue.

1. Enable the debug output in the VS Code settings under 'RPGLE language tools'

<img width="734" alt="image" src="https://user-images.githubusercontent.com/3708366/191402998-e8535eb1-2d1d-45fd-ada5-3816a098a590.png">

2. Restart VS Code and cause the issue to happen again.
3. A new message will appear asking you to display the debug info

<img width="526" alt="image" src="https://user-images.githubusercontent.com/3708366/191403094-0fc2797a-d59a-4b3d-9994-7915efff4a63.png">

4. Copy the JSON to your clipboard and share it with us here

<img width="472" alt="image" src="https://user-images.githubusercontent.com/3708366/191403201-5accb182-c94d-4e6f-a646-7eb928eca7fb.png">

**Comment by worksofliam** [comment]
The copybooks can be in fixed with no problem. The linter only runs on active source (which is free).

Let me take a look at this and get back to you.

**Comment by worksofliam** [comment]
@dennisnel I have just release 0.10.7 which may solve your problem. Give it a go and let me know. Thanks!

**Comment by dennisnel** [comment]
Hi @worksofliam , I was able to open the member that was previously giving an error without any issues.

But when I tried a new RPG member I got an error again.

I have included the source as well as all the /Include members in the zip file.

I did notice that when I opened them to get the outline view only the CATEMSTPF.rpgle copybook produced results.

I hope this helps

[RPGLE Linter Error.zip](https://github.com/halcyon-tech/vscode-rpgle/files/9624742/RPGLE.Linter.Error.zip)
